cblox/include/cblox/core/submap_collection.h

```diff
@@ -0,0 +1,138 @@
+                       const SubmapID new_submap_id);
+
+  // Gets a const pointer to a raw submap
+  // NOTE(alexmillane): This function hard fails when the submap doesn't
+  // exist... This puts the onus on the caller to call exists() first. I don't
+  // like this but I can't see a solution.
+  const SubmapType &getSubMap(const SubmapID submap_id) const;
+  // Note(alexmillane): Unlike the above this function returns a nullptr when
+  // the map doesn't exist. No hard crash.
+  typename SubmapType::ConstPtr getSubMapConstPtrById(
+      const SubmapID submap_id) const;
+  // A list of the submaps
+  const std::vector<typename SubmapType::Ptr> getSubMaps() const;
+
+  // Interactions with the active submap
+  const SubmapType &getActiveSubMap() const;
+  const Transformation &getActiveSubMapPose() const;
+  const SubmapID getActiveSubMapID() const;
+
+  // Access the tsdf_map member of the active submap
+  TsdfMap::Ptr getActiveTsdfMapPtr();
+  const TsdfMap &getActiveTsdfMap() const;
+
+  // Interacting with the submap poses
+  bool setSubMapPose(const SubmapID submap_id, const Transformation &pose);
+  void setSubMapPoses(const TransformationVector &transforms);
+  bool getSubMapPose(const SubmapID submap_id, Transformation *pose_ptr) const;
+  void getSubMapPoses(TransformationVector* submap_poses) const;
+
+  // Generate the ESDF for the submap with the given ID. This method is only
+  // available if the SubmapType is equal to or derived from TsdfEsdfSubmap.
+  template <class ST = SubmapType>
+  typename std::enable_if<std::is_base_of<TsdfEsdfSubmap, ST>::value,
+                          bool>::type
+  generateEsdfById(const SubmapID submap_id) {
+    // Find the submap corresponding to the given ID
+    const auto submap_ptr_it = id_to_submap_.find(submap_id);
+    if (submap_ptr_it != id_to_submap_.end()) {
+      // Generate the ESDF from the TSDF
+      submap_ptr_it->second->generateEsdf();
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  // Clears the collection, leaving an empty map
+  void clear() { id_to_submap_.clear(); }
+
+  // Size information
+  bool empty() const { return id_to_submap_.empty(); }
+  size_t size() const { return id_to_submap_.size(); }
+  size_t num_patches() const { return id_to_submap_.size(); }
+  FloatingPoint block_size() const {
+    return (id_to_submap_.begin()->second)->block_size();
+  }
+  size_t getNumberAllocatedBlocks() const;
+
+  // Returns the config of the submaps
+  const typename SubmapType::Config &getConfig() const {
+    return submap_config_;
+  }
+
+  // Save the collection to file
+  bool saveToFile(const std::string &file_path) const;
+  void getProto(TsdfSubmapCollectionProto *proto) const;
+
+  // Fusing the submap pairs
+  void fuseSubmapPair(const SubmapIdPair &submap_id_pair);
+
+  // Flattens the collection map down to a normal TSDF map
+  TsdfMap::Ptr getProjectedMap() const;
+
+  // KEYFRAME RELATED FUNCTIONS.
```

Comment:
I'm just gonna remove these now. We don't want the keyframe related stuff in the library.